src/uct/cuda/gdr_copy/gdr_copy_md.c

```diff
@@ -235,17 +247,46 @@ uct_gdr_copy_query_md_resources(uct_component_t *component,
                                             num_resources_p);
 }
 
-static void uct_gdr_copy_md_close(uct_md_h uct_md)
+static void uct_gdr_copy_md_release(uct_gdr_copy_md_t *md)
 {
-    uct_gdr_copy_md_t *md = ucs_derived_of(uct_md, uct_gdr_copy_md_t);
     int ret;
 
+    if (md->rcache != NULL) {
+        ucs_rcache_destroy(md->rcache);
+    }
+
     ret = gdr_close(md->gdrcpy_ctx);
     if (ret) {
         ucs_warn("failed to close gdrcopy. ret:%d", ret);
     }
+}
+
+typedef struct {
+    pthread_mutex_t   lock;
```

Comment:
why do we need a lock?

Reply:
need to synchronize different contexts

Reply:
is it because ucp memory routines are not protected with a global lock on ucp worker?

Reply:
single md means md shared between multiple ucp contexts, it's before workers

Reply:
pls move the typedef and global var declaration to the beginning of this C file